# TargetGenes
This code was my final assignment within my MBINF program.  Here I have used python to take raw rnaSeq data and utilize machine learning methods along with PCA to identify lists of genes that may be targeted by the Wnt signaling pathway weighted by importance.  This code can be utilized with any rnaSeq data that has been cleaned and normalized for any purpose.  

The .py files involve the code for creating the top gene lists, and the .R files involve the code for cleaning my data.  Data used for this was garnered from publically available data from the following studies:

1.	Bell, I., Khan, H., Stutt, N., Horn, M., Hydzik, T., Lum, W., Rea, V., Clapham, E., Hoeg, L., & Van Raay, T. J. (2024). Nkd1 functions downstream of Axin2 to attenuate Wnt signaling. Molecular Biology of the Cell, 35(7), ar93. https://doi.org/10.1091/mbc.E24-02-0059-T 
   
2.	Mukherjee, S., Luedeke, D. M., McCoy, L., Iwafuchi, M., & Zorn, A. M. (2022). SOX transcription factors direct TCF-independent WNT/β-catenin responsive transcription to govern cell fate in human pluripotent stem cells. Cell Reports, 40(8), 111247. https://doi.org/10.1016/j.celrep.2022.111247 
   
3.	Nakagawa, S., Yamaguchi, K., Takane, K., Tabata, S., Ikenoue, T., & Furukawa, Y. (2024). Wnt/β-catenin signaling regulates amino acid metabolism through the suppression of CEBPA and FOXA1 in liver cancer cells. Communications Biology, 7(1), 510. https://doi.org/10.1038/s42003-024-06202-9 
   
4.	Pagella, P., Söderholm, S., Nordin, A., Zambanini, G., Ghezzi, V., Jauregi-Miguel, A., & Cantù, C. (2023). The time-resolved genomic impact of Wnt/β-catenin signaling. Cell Systems, 14(7), 563-581.e7. https://doi.org/10.1016/j.cels.2023.06.004 

